### PR TITLE
Ignore warnings for destructor in ViewOfViews test

### PR DESCRIPTION
**Pull request by masterleinad**
This fixes the failing `CUDA-11.6-NVCC-DEBUG'` CI.

### CODE REVIEW

**Comment by masterleinad** [comment]
Retest this please.

**Comment by masterleinad** [comment]
Retest this please.

File: core/unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -300,6 +300,12 @@ foreach(Tag Threads;Serial;OpenMP;Cuda;HPX;OpenMPTarget;OpenACC;HIP;SYCL)
       list(APPEND ${Tag}_SOURCES2A ${file})
     endforeach()
 
+    # disable a warning about calling a host-only destructor in the ViewOfView test
+    # if we use nvcc_wrapper directly.
+    if(KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA AND NOT INTERNAL_USE_COMPILER_LAUNCHER)
+      set_source_files_properties(${dir}/Test${Tag}_ViewOfViews.cpp PROPERTIES COMPILE_FLAGS "-Xcudafe --diag_suppress=20014")
```
Comment (dalg24) [inline core/unit_test/CMakeLists.txt:306]: I would prefer a pragma push restricted to the NVCC versions that need it

**Comment by masterleinad** [comment]
> Do you understand why this is only showing up in that one CUDA 11.6 debug build?
> It looks like it is being emitted from that `ViewValueFunctor::functor_instantiate_workaround()` function.

No, but it seems reasonable for `nvcc` to complain about calling a host-only destructor from a functor (`__host__ __device__` function).

**Comment by masterleinad** [comment]
We found a better workaround with @dalg24 and @crtrott.

**Pull request closed without merging**